modules/apps/portal-workflow/portal-workflow-kaleo-service/src/main/java/com/liferay/portal/workflow/kaleo/internal/upgrade/v1_4_1/KaleoDefinitionVersionUpgradeProcess.java

```diff
@@ -186,6 +196,62 @@ protected void upgradeKaleoDefinitionVersion() throws Exception {
 				}
 			}
 
+			DBInspector dbInspector = new DBInspector(connection);
+
+			DatabaseMetaData metaData = connection.getMetaData();
+
+			ResultSet metaDataResultSet = metaData.getTables(
+				dbInspector.getCatalog(), dbInspector.getSchema(),
+				"KaleoDraftDefinition", new String[] {"TABLE"});
+
+			if (metaDataResultSet.next()) {
+				PreparedStatement preparedStatement3 =
```

Comment:
@jesseyeh-liferay does the upgrade _com.liferay.portal.workflow.kaleo.designer.web.internal.upgrade.v1_0_1.KaleoDefinitionVersionUpgradeProcess_ make sense after adding this code here?

Reply:
@achaparro 
I tested this by using a dummy upgrade step and found that the upgraded KaleoDefinitionVersion table was identical. See https://github.com/achaparro/liferay-portal/pull/937/commits/0d838e469b120db2b6a4249171389b641507a76a.